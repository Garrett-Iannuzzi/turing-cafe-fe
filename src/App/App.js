import React, { Component } from 'react';
import ReservationContainer from '../ReservationContainer/ReservationContainer';
import Form from '../Form/Form';
import './App.css';

class App extends Component {
  constructor() {
    super()
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(reservations => this.setState({ reservations }))
  }

  handleAddReservation = (newRes) => {
    this.setState({ reservations: [...this.state.reservations, newRes] })
  }

  render() {
    return (
      <main className='app'>
        <h1 className='app-title'>Turing Cafe!</h1>
        <Form
          handleAddReservation={this.handleAddReservation}
        />
        <ReservationContainer
          reservations={this.state.reservations}
        />
      </main>
    )
  }
}

export default App;
