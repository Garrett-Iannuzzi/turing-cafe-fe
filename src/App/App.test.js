import React from 'react';
import { shallow } from 'enzyme';
import App from '../App/App';

describe('App', () => {

  it('Should match the snapshot with the correct data', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Should update state when handleAddReservation is called', () => {
    const wrapper = shallow(<App />)
    const mockReservation = { id: 20, name: 'Nick', date: '01/01', time: '7:00', number: 2 }
    wrapper.setState({ reservations: [] });
    wrapper.instance().handleAddReservation(mockReservation);

    expect(wrapper.state('reservations')).toEqual([ mockReservation ])
  });
});
