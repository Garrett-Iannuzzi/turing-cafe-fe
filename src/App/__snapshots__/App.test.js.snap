// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot with the correct data 1`] = `
<main
  className="app"
>
  <h1
    className="app-title"
  >
    Turing Cafe!
  </h1>
  <Form
    handleAddReservation={[Function]}
  />
  <ReservationContainer
    reservations={Array []}
  />
</main>
`;
